draft-pda-protocol.md

```diff
@@ -950,31 +956,68 @@ indefinitely until a suitable output is found.
 
 [TODO: Define Hits-specific protocol messages.]
 
-# System design
+# Operational Considerations {#operational-capabilities}
+
+Prio has inherent constraints derived from the tradeoff between privacy
+guarantees and computational complexity. These tradeoffs influence how
+applications may choose to utilize services implementing the specification.
 
-[[OPEN ISSUE: This section seems like a catch-all for things not in other
-sections. Perhaps there is a natural home for aggregator discovery, share
-uploading, open issues, and system parameters?]]
+## Protocol participant capabilities {#entity-capabilities}
 
-## Aggregator discovery
+The design in this document has different assumptions and requirements for
+different protocol participants, including clients, aggregators, and
+collectors. This section describes these capabilities in more detail.
 
-[[OPEN ISSUE: writeme]]
+### Client capabilities
 
-## Share uploading
+Clients have limited capabilities and requirements. Their only inputs to the protocol
+are (1) the PDAParam structure configured out of band and (2) a corresponding value to be
+submitted in a report. Clients are not expected to store any state across any upload
+flows, nor are required to implement any sort of report retry mechanism should it fail.
+By design, the protocol in this document is robust against individual client upload
+failures since the protocol output is an aggregate over all inputs.
 
-[[OPEN ISSUE: writeme]]
+### Aggregator capabilities
 
-## Open questions and system parameters {#questions-and-params}
+Helpers and leaders have different operational requirements. The design in this
+document assumes an operationally competent leader, i.e., one that has no storage
+or computation limitations or constraints, and minimalistic helper, i.e., one that
+has computation, bandwidth, and storage constraints. By design, leaders must be
+at least as capable as helpers, where helpers are generally required to:
 
-[[OPEN ISSUE: discuss batch size parameter and thresholds]]
-[[OPEN ISSUE: discuss f^ leakage differences from [GB17]]]
+- Support the collect protocol, which includes verifying and aggregating
+  sets of reports in a given batch; and
+- Publish and manage an HPKE configuration that can be used for the upload protocol;
 
+In addition, for each PDAParam instance, helpers are required to:
 
-# Operational Considerations
+- Implement some form of batch-to-report and intra-batch replay mitigation storage,
+  which includes some way of tracking batch report size, with optional support for
+  state offloading.
 
-Prio has inherent constraints derived from the tradeoff between privacy
-guarantees and computational complexity. These tradeoffs influence how
-applications may choose to utilize services implementing the specification.
+[OPEN ISSUE: describe intra- and inter-batch replay attacks in the security considerations]
```

Comment:
This will be resolved (at least partially) by #90.

Reply:
Yep, noted it as such.